src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -370,3 +370,188 @@ def get_token(self, *scopes, **kwargs):  # pylint: disable=unused-argument
+    from requests import Request, Session
+
+    try:
+        with open(script_path, 'r', encoding='utf-8') as f:
+            script_content = f.read()
+    except FileNotFoundError:
+        raise CLIError(f"Script file not found: {script_path}")
+    except Exception as ex:
+        raise CLIError(f"Error reading script file: {ex}")
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+    payload = {
+        "azcli_script": script_content,
+        "subscription_id": subscription_id
+    }
+
+    request_completed = threading.Event()
+
+    def rotating_progress():
+        """Simulate a rotating progress indicator, similar to the one displayed during long-running operations.
+        """
+        chars = ["|", "\\", "/", "-"]
+        idx = 0
+        while not request_completed.is_set():
+            sys.stderr.write(f"\r{chars[idx % len(chars)]} Running")
+            sys.stderr.flush()
+            idx += 1
+            time.sleep(0.2)
+        sys.stderr.write("\r" + " " * 20 + "\r")
+        sys.stderr.flush()
+
+    try:
+        FUNCTION_APP_URL = "https://azcli-script-insight.azurewebsites.net"
+        resource = cmd.cli_ctx.cloud.endpoints.active_directory_resource_id
+        profile = Profile(cli_ctx=cmd.cli_ctx)
+
+        try:
+            token_result = profile.get_raw_token(resource, subscription=subscription_id)
+            token_info, _, _ = token_result
+            token_type, token, _ = token_info
+        except Exception as token_ex:
+            request_completed.set()
+            raise CLIError(f"Failed to get authentication token: {token_ex}")
+
+        headers_dict = {}
+        headers_dict['Authorization'] = '{} {}'.format(token_type, token)
+        headers_dict['Content-Type'] = 'application/json'
+
+        progress_thread = threading.Thread(target=rotating_progress)
+        progress_thread.daemon = True
+        progress_thread.start()
+
+        session = Session()
+        req = Request(method="POST", url=f"{FUNCTION_APP_URL}/api/what_if_preview",
+                      headers=headers_dict, data=json.dumps(payload))
+        prepared = session.prepare_request(req)
+        response = session.send(prepared)
+        request_completed.set()
+
+        progress_thread.join(timeout=0.5)
+
+    except Exception as ex:
+        request_completed.set()
+        if 'progress_thread' in locals():
+            progress_thread.join(timeout=0.5)
+        raise CLIError(f"Failed to connect to the what-if service: {ex}")
+
+    try:
+        raw_results = response.json()
+    except ValueError as ex:
+        raise CLIError(f"Failed to parse response from what-if service: {ex}")
+
+    what_if_result = raw_results.get('what_if_result', {})
+    what_if_operation_result = _convert_json_to_what_if_result(what_if_result)
+
+    if no_pretty_print:
+        return what_if_result
+
+    print(format_what_if_operation_result(what_if_operation_result, cmd.cli_ctx.enable_color))
+    return what_if_result
```

Comment:
```suggestion
    try:
        raw_results = response.json()
    except ValueError as ex:
        raise CLIError(f"Failed to parse response from what-if service: {ex}")

    success = raw_results.get('success')
    if success is False:
        return raw_results
    elif success is True:
        what_if_result = raw_results.get('what_if_result', {})
        what_if_operation_result = _convert_json_to_what_if_result(what_if_result)
        if no_pretty_print:
            return what_if_result
        print(format_what_if_operation_result(what_if_operation_result, cli_ctx.enable_color))
        return what_if_result
    else:
        raise CLIError(f"Unexpected response from what-if service, got: {raw_results}")
```
If the What-If service reports an error, we should also output the error message.
Here’s a comparison of the changes before and after the modification:
<img width="1386" height="214" alt="image" src="https://github.com/user-attachments/assets/5e554395-559b-49b3-9e4c-0af3591317b6" />

<img width="1768" height="332" alt="image" src="https://github.com/user-attachments/assets/2f96f9b2-f4f9-41eb-bd87-f4dec9f5e278" />